packages/react/src/file-upload/FileUpload.tsx

```diff
@@ -282,22 +348,63 @@ export const FileUpload = forwardRef<HTMLInputElement, FileUploadProps>(
 									Select {filesPlural}
 								</Button>
 							</Flex>
-							{showFileLists && (
+							{areFileListsVisible && (
 								<Stack gap={0.5}>
-									<Text color="muted">{fileSummaryText}</Text>
+									{Boolean(allRejections.length) && (
+										<SectionAlert
+											title="The following files could not be selected"
+											tone="error"
+											onClose={clearErrors}
+										>
+											<Text as="p">
+												These files did not meet upload requirements, were not
+												listed, and will not be uploaded.
+											</Text>
+											<UnorderedList>
+												{allRejections.map((file) => {
```

Comment:
What do you think about using the singular of `allRejections`, i.e. `rejection` instead of `file`? That would save us the the awkward `file.file` on lines 366, 368, and 371.